src/comparator/service_comparator.py

```diff
@@ -94,28 +119,33 @@ def _compareRpcMethods(self, service_original, service_update):
                     FindingCategory.METHOD_SERVER_STREAMING_CHANGE, "", msg, True
                 )
             # 6.7 The paginated response of an RPC method is changed.
-            if self._isPaginatedResponse(method_original) != self._isPaginatedResponse(
-                method_update
-            ):
+            if self._isPaginatedResponse(
+                method_original, messages_map_original
+            ) != self._isPaginatedResponse(method_update, messages_map_update):
                 msg = "The paginated response of method {} is changed".format(name)
                 FindingContainer.addFinding(
                     FindingCategory.METHOD_PAGINATED_RESPONSE_CHANGE, "", msg, True
                 )
 
-    def _isPaginatedResponse(self, method: MethodDescriptorProto) -> bool:
+    def _isPaginatedResponse(
+        self, method: MethodDescriptorProto, messages_map: Dict[str, DescriptorProto]
+    ) -> bool:
         # (AIP 158) The response must not be a streaming response.
         if method.server_streaming:
             return False
-        responseMsg = method.output_type
-        # API must provide a `string next_page_token` field.
-        if (
-            "next_page_token" not in responseMsg.fields_by_name
-        ) or responseMsg.fields_by_name["next_page_token"].type != 9:
-            return False
-        # The field containing pagination results should be the first
-        # field in the message and have a field number of 1.
-        # It should be a repeated field containing a list of resources
-        # constituting a single page of results.
-        if responseMsg.fields_by_number[1].label != 3:
-            return False
-        return True
+        # Short message name e.g. .example.v1.FooRequest -> FooRequest
+        response_message_name = method.output_type.rsplit(".", 1)[-1]
+        response_message = messages_map[response_message_name]
+        next_page_token = False
+        pagination_result = False
+        for f in response_message.field:
+            # API must provide a `string next_page_token` field.
+            if f.name == "next_page_token" and f.type == 9:
+                next_page_token = True
+            # The field containing pagination results should be the first
+            # field in the message and have a field number of 1.
+            # It should be a repeated field containing a list of resources
+            # constituting a single page of results.
+            if f.number == 1 and f.label == 3:
+                pagination_result = True
+        return next_page_token and pagination_result
```

Comment:
Take a look at the logic in https://github.com/googleapis/gapic-generator-python/blob/master/gapic/schema/wrappers.py#L799 and make sure that it agrees with what you have here.

Reply:
Thanks for the resource! another TWIL :) that we can call a function with arguments by putting them in a tuple (link above). I stole the implementation from python micro, and add one more check that the repeated field in the response should numbered 1.  Thanks!